bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusHandlerFactory.java

```diff
@@ -43,21 +50,72 @@
 
     private final Map<ThingUID, @Nullable ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();
 
+    private @Nullable HttpService httpService;
+    private @Nullable StorageService storageService;
+    private @Nullable AccountsServlet accountsServlet;
+
     private static final Collection<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Arrays.asList(THING_TYPE_SENSEGUARD,
             THING_TYPE_SENSE, THING_TYPE_BRIDGE_ACCOUNT);
 
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, policy = ReferencePolicy.DYNAMIC)
+    protected void setHttpService(HttpService httpService) {
+        this.httpService = httpService;
+    }
+
+    protected void unsetHttpService(HttpService httpService) {
+        this.httpService = null;
+    }
+
+    @Reference
+    protected void setStorageService(StorageService storageService) {
+        this.storageService = storageService;
+    }
+
+    protected void unsetStorageService(StorageService storageService) {
+        this.storageService = null;
+    }
+
     @Override
     public boolean supportsThingType(ThingTypeUID thingTypeUID) {
         return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
     }
 
+    @Override
+    protected void activate(ComponentContext componentContext) {
+        super.activate(componentContext);
+        HttpService httpService = this.httpService;
+        if (accountsServlet == null && httpService != null) {
+            accountsServlet = new AccountsServlet(httpService);
+        }
+    }
+
+    @Override
+    protected void deactivate(ComponentContext componentContext) {
+        AccountsServlet accountsServlet = this.accountsServlet;
+        this.accountsServlet = null;
+        if (accountsServlet != null) {
+            accountsServlet.dispose();
+        }
+        super.deactivate(componentContext);
+    }
+
     @Override
     protected @Nullable ThingHandler createHandler(Thing thing) {
+        HttpService httpService = this.httpService;
+        if (httpService == null) {
+            return null;
+        }
+        StorageService storageService = this.storageService;
+        if (storageService == null) {
+            return null;
+        }
+
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
 
         if (THING_TYPE_BRIDGE_ACCOUNT.equals(thingTypeUID)) {
-            GroheOndusAccountHandler handler = new GroheOndusAccountHandler((Bridge) thing);
-            registerDeviceDiscoveryService(handler);
+            GroheOndusAccountHandler handler = new GroheOndusAccountHandler((Bridge) thing, httpService,
```

Comment:
Use the classloader of your bundle. You get that via `FrameworkUtils.getBundle().getBundleContext().classloader()` or so.

Reply:
These methods does not exist, however, I found a way to get the classloader from the bundle.